gobblin-modules/gobblin-metadata/src/main/java/gobblin/metadata/types/GlobalMetadata.java

```diff
@@ -41,22 +45,44 @@
   private static final Logger log = LoggerFactory.getLogger(GlobalMetadata.class);
   private static final ObjectMapper objectMapper = new ObjectMapper();
   private static final JsonFactory jsonFactory = new JsonFactory();
+  private static final String EMPTY_ID = "0";
 
   @JsonProperty("dataset")
   private final Map<String, Object> datasetLevel;
 
   @JsonProperty("file")
   private final Map<String, Map<String, Object>> fileLevel;
 
+  @JsonProperty("id")
+  private String cachedId;
+
+  private transient boolean markedImmutable;
+
   private final static String DATASET_URN_KEY = "Dataset-URN";
```

Comment:
Hmm, I can stash them in an enum instead to make the code a little cleaner. What do you think adding a new explicit MetadataName type does? Possible extension / self-documentation later?